app/controllers/vendor_api/confirm_candidate_enrolment_controller.rb

```diff
@@ -13,5 +17,27 @@ def confirm
         }
       end
     end
+
+  private
+
+    def check_metadata!
```

Comment:
What woud you think of extracting the metadata bit into a object with validations? Eg:

```rb
class Metadata
  attr_accessor :timestamp
  validates_presence_of :timestamp
  ...
end
```

- This will cover more cases - currently a `nil` timestamp won't trigger an error, which is not really what we want
- Allow us to verify that the date is a date etc
- It'll allow you to use the `errors` in the error message - the current error message isn't explaining the exact thing that is wrong

Reply:
We YAGNI'd ourselves out of doing this, but that you think it's a good idea on review is enough to tip the balance: will add :+1:

Reply:
done  ;)